### PR TITLE
add ability to stop/start the websocket connection

### PR DESCRIPTION
**Pull request by camelpac**
In this PR we bring back the ability to start/stop a ws connection giving algorithms the flexabilty to run longer periods of time even when market is closed.

### CODE REVIEW

File: examples/websockets/streamconn_on_and_off.py [hunk 1]
```diff
@@ -48,10 +49,11 @@ def consumer_thread():
...
     loop = asyncio.get_event_loop()
+    pool = ThreadPoolExecutor(1)
 
     while 1:
         try:
-            threading.Thread(target=consumer_thread).start()
+            pool.submit(consumer_thread)
             time.sleep(20)
             loop.run_until_complete(conn.stop_ws())
             time.sleep(20)
```
[the remark this answers is not in the document]
Comment (camelpac) [reply examples/websockets/streamconn_on_and_off.py:59]: ok

[the remark this answers is not in the document]
Comment (camelpac) [reply examples/websockets/streamconn_on_and_off.py:59]: ok

File: alpaca_trade_api/stream.py [hunk 2]
```diff
@@ -663,3 +692,13 @@ async def stop_ws(self):
         if self._crypto_ws:
             log.info("Stopping the crypto data websocket connection")
             await self._crypto_ws.stop_ws()
+
```
[the remark this answers is not in the document]
Comment (gnvk) [reply alpaca_trade_api/stream.py:696]: I meant that `if self._trading_ws:` should be `if self._trading_ws._ws`, shouldn't it?

File: examples/websockets/streamconn_on_and_off.py [hunk 3]
```diff
@@ -48,12 +49,18 @@ def consumer_thread():
...
-        except:
+        except KeyboardInterrupt:
+            print("Interrupted execution by user")
+            loop.run_until_complete(conn.stop_ws())
+            exit(0)
+        except Exception:
+            # let the execution continue
```
Comment (gnvk) [inline examples/websockets/streamconn_on_and_off.py:65]: At the very least log it! But I'd prefer letting this through.

File: alpaca_trade_api/stream.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (camelpac) [reply alpaca_trade_api/stream.py:696]: ok check that too.

File: examples/websockets/streamconn_on_and_off.py (hunk 3 above)
Comment (camelpac) [reply examples/websockets/streamconn_on_and_off.py:67]: ok

**Review by gnvk — APPROVED** [review]
LGTM

**Pull request merged**